encrypt/encrypt.go

```diff
@@ -27,14 +36,271 @@ var ArgHelp = `
+		ChecksumFileEncSha256.WriteString(
+			fmt.Sprintf("%s %s\n", hashes.encryptedSha256, file.encrypted))
+	}
+}
+
+// Checks that all the input files exists, and are readable
+func checkFiles(files []encryptionFileSet) error {
+	log.Info("Checking files")
+	for _, file := range files {
+		// check that the input file exists
+		fileInfo, err := os.Stat(file.unencrypted)
+		if err != nil {
+			return err
+		}
+		// check that the input file isn't a directory
+		if fileInfo.IsDir() {
+			return fmt.Errorf("%s is a directory", file.unencrypted)
+		}
+		// check that the output file doesn't exist
+		_, err = os.Stat(file.encrypted)
+		if err == nil {
+			return fmt.Errorf("outfile %s already exists", file.encrypted)
+		}
+	}
+	return nil
+}
+
+// Calculates md5 and sha256 hashes for the unencrypted and encrypted file
+func calculateHashes(fileSet encryptionFileSet) (*hashSet, error) {
+
+	hashes := hashSet{"", "", "", ""}
+
+	// open infile
+	inFile, err := os.Open(fileSet.unencrypted)
+	if err != nil {
+		return nil, err
+	}
+	defer inFile.Close()
+
+	// unencrypted md5
+	md5Hash := md5.New()
+	_, err = io.Copy(md5Hash, inFile)
+	if err != nil {
+		return nil, err
+	}
+	hashes.unencryptedMd5 = hex.EncodeToString(md5Hash.Sum(nil))
+
+	// reset file
+	_, err = inFile.Seek(0, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	// unencrypted sha256
+	shaHash := sha256.New()
+	_, err = io.Copy(shaHash, inFile)
+	if err != nil {
+		return nil, err
+	}
+	hashes.unencryptedSha256 = hex.EncodeToString(shaHash.Sum(nil))
+
+	// open outfile
+	outFile, err := os.Open(fileSet.encrypted)
+	if err != nil {
+		return nil, err
+	}
+	defer outFile.Close()
+
+	// encrypted md5
+	md5Hash.Reset()
+	_, err = io.Copy(md5Hash, outFile)
+	if err != nil {
+		return nil, err
+	}
+	hashes.encryptedMd5 = hex.EncodeToString(md5Hash.Sum(nil))
+
+	// reset file
+	_, err = outFile.Seek(0, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	// encrypted sha256
+	shaHash.Reset()
+	_, err = io.Copy(shaHash, outFile)
+	if err != nil {
+		return nil, err
+	}
+	hashes.encryptedSha256 = hex.EncodeToString(shaHash.Sum(nil))
+
+	return &hashes, nil
+}
+
+// Reads a public key file from a file using the crypt4gh keys module
+func readPublicKey(filename string) (*[32]byte, error) {
+	log.Info("Reading Public key file")
+	file, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	publicKey, err := keys.ReadPublicKey(file)
+	if err != nil {
+		return nil, err
+	}
+	return &publicKey, nil
+}
+
+// Generates a crypt4gh key pair, returning only the private key, as the
+// public key isn't needed for encryption.
+func generatePrivateKey() (*[32]byte, error) {
+	log.Info("Generating encryption key")
+
+	_, privateKey, err := keys.GenerateKeyPair()
+	if err != nil {
+		return nil, err
+	}
+	return &privateKey, nil
+}
+
+// Encrypts the data from `filename` into `outFilename` for the given `pubKey`,
+// using the given `privateKey`.
```

Comment:
```suggestion
// using the generated `privateKey`.
```

please discard if I misunderstand sth

Reply:
in this context, it's the private key that's given to the function, so it's correct. :) But I'll see if I can be more clear in the future